src/connectionmanager.js

```diff
@@ -567,36 +458,22 @@ export default class ConnectionManager {
 
             function onLocalUserDone(e) {
 
-                const image = getImageUrl(localUser);
-
-                resolve({
-                    localUser,
-                    name: connectUser ? connectUser.Name : (localUser ? localUser.Name : null),
-                    imageUrl: image.url,
-                    supportsImageParams: image.supportsParams,
-                    connectUser
-                });
-            }
-
-            function onEnsureConnectUserDone() {
-
                 if (apiClient && apiClient.getCurrentUserId()) {
                     apiClient.getCurrentUser().then(u => {
                         localUser = u;
-                        onLocalUserDone();
+                        const image = getImageUrl(localUser);
 
+                        resolve({
+                            localUser,
+                            name: (localUser ? localUser.Name : null),
+                            imageUrl: image.url,
+                            supportsImageParams: image.supportsParams,
+                        });
                     }, onLocalUserDone);
```

Comment:
remove the recursive call on failure

Reply:
you're saying remove this entirely

Reply:
I might be wrong, but isn't onLocalUserDone called inside itself here as a failure condition?